src/shared/components/simple-select/SimpleSelect.tsx

```diff
@@ -0,0 +1,130 @@
+ * limitations under the License.
+ */
+
+import React, { type HTMLAttributes } from 'react';
+import classNames from 'classnames';
+import styles from './SimpleSelect.scss';
+
+type HTMLSelectProps = HTMLAttributes<HTMLSelectElement>;
+
+// This is just a simple wrapper for the native HTML select element.
+// The purpose of this React component is to style the select button,
+// since it's the only part of the native select element that can be styled.
+
+// NOTE: when interacting with SimpleSelect component,
+// pass your event handler to the onInput rather than the onChange property,
+// because onInput fires immediately, while the onChange fires when the select gets unfocussed.
+
+export type Option = {
+  value: string;
+  label: string;
+};
+
+export type OptionGroup = {
+  title?: string;
+  options: Option[];
+};
+
+type OptionsSpecificProps = {
+  options: Option[];
+  title?: string;
+};
+
+type OptionGroupsSpecificProps = {
+  optionGroups: OptionGroup[];
+};
+
+type CommonProps = {
+  placeholder?: string;
+};
+
+type OptionsSelectProps = CommonProps & OptionsSpecificProps;
+type OptionGroupsSelectProps = CommonProps & OptionGroupsSpecificProps;
+
+type SimpleSelectAdapterProps = HTMLSelectProps &
+  (OptionsSelectProps | OptionGroupsSelectProps);
+
+export type SimpleSelectProps = HTMLSelectProps &
+  OptionGroupsSelectProps &
+  CommonProps;
+
+const SimpleSelect = (props: SimpleSelectProps) => {
+  const { optionGroups, placeholder, ...rest } = props;
+
+  const selectClassnames = classNames(styles.select, props.className);
+
+  if (optionGroups.length === 1) {
+    return (
+      <div className={selectClassnames}>
+        <select
+          className={styles.selectResetDefaults}
+          defaultValue=""
+          {...rest}
+        >
+          {placeholder && (
+            <option value="" hidden={true}>
+              {placeholder}
+            </option>
+          )}
+          {optionGroups[0].options.map((option, key) => (
+            <option key={key} value={option.value}>
+              {option.label}
+            </option>
+          ))}
+        </select>
+      </div>
+    );
+  }
+
+  return (
+    <div className={selectClassnames}>
+      <select className={styles.selectResetDefaults} defaultValue="" {...rest}>
+        {placeholder && (
+          <option value="" hidden={true}>
+            {placeholder}
+          </option>
+        )}
+        {optionGroups.map((optionGroup, optionGroupKey) => {
+          return (
+            <optgroup label={optionGroup.title} key={optionGroupKey}>
+              {optionGroup.options.map((option, optionKey) => (
+                <option key={optionKey} value={option.value}>
+                  {option.label}
+                </option>
+              ))}
+            </optgroup>
+          );
+        })}
+      </select>
+    </div>
+  );
+};
+
+const SimpleSelectAdapter = (props: SimpleSelectAdapterProps) => {
```

Comment:
Reading through the code, I don't even see the point in this adapter component. What if you passed the props directly to the Select, and if the props contain an `optionGroups` field, then you build the Select one way, whereas if it contains the `options` field, you build it another way?

BTW, you shouldn't need to write `if ((props as OptionGroupsSelectProps).optionGroups)`. Typescript should be able to infer the type of props if you write `if ('optionGroups' in props)` etc.

Reply:
That's done